### PR TITLE
feat/version-per-typeurl

### PR DESCRIPTION
**Pull request by roivaz**
So far all resource types shared the same version. This causes that when one resource type (for example the listeners) has changes, the version changes for all other resource types too and envoy clients refresh all of them, which can have an impact on performance. This PR changes this and version is now done in a per type basis.

**Note for reviewers**: I have disabled the self-healing functionality in this PR as this change breaks it. An upcoming PR will reimplement the self-healing. I will not merge this PR until the other one is ready for merging toavoid leaving `main` broken.

/kind feature
/priority important-soon
/hold

### CODE REVIEW

File: pkg/discoveryservice/manager.go [hunk 1]
```diff
@@ -92,7 +91,8 @@ func (dsm *Manager) Start(ctx context.Context) {
 			ClientAuth:   tls.RequireAndVerifyClientCert,
 			ClientCAs:    loadCA(dsm.CACertificatePath, setupLog),
 		},
-		rollback.OnError(mgr.GetClient()),
+		// rollback.OnError(mgr.GetClient()),
```
Comment (slopezz) [inline pkg/discoveryservice/manager.go:94]: is this comment on purpose?

Comment (roivaz) [reply pkg/discoveryservice/manager.go:94]: Yes, is the temporary disable of the self-healing functionality.

Comment (slopezz) [reply pkg/discoveryservice/manager.go:94]: nice!

**Pull request merged**